server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/chatDb/chatDb.ts

```diff
@@ -844,4 +844,15 @@ export class ChatDatabase {
     setModelId(modelId: string | undefined): void {
         this.updateSettings({ modelId: modelId === '' ? undefined : modelId })
```

Comment:
I think here you'll have to do what you did below, otherwise it might reset pairProgrammingMode?

Reply:
oh in that case why do we need to do the existing settings check for setPairProgrammingMode?